include/openenclave/edger8r/common.h

```diff
@@ -116,44 +116,44 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * Also copy the contents of the corresponding in-out pointer in the input
  * buffer.
  */
-#define OE_COPY_AND_SET_IN_OUT_POINTER(argname, argsize)                       \
-    if (pargs_in->argname)                                                     \
-    {                                                                          \
-        uint8_t* _p_in = (uint8_t*)pargs_in->argname;                          \
-        *(uint8_t**)&pargs_in->argname = output_buffer + output_buffer_offset; \
-        OE_ADD_SIZE(output_buffer_offset, (size_t)argsize);                    \
-        if (output_buffer_offset > output_buffer_size)                         \
-        {                                                                      \
-            _result = OE_BUFFER_TOO_SMALL;                                     \
-            goto done;                                                         \
-        }                                                                      \
-        memcpy(pargs_in->argname, _p_in, (size_t)(argsize));                   \
+#define OE_COPY_AND_SET_IN_OUT_POINTER(argname, argsize, argtype)            \
+    if (pargs_in->argname)                                                   \
+    {                                                                        \
+        argtype _p_in = (argtype)pargs_in->argname;                          \
```

Comment:
Are we reaching here for foreign arrays.
Given:
typedef int my_array_t[5];

void foo([in, isary] my_array_t input);

We don't know what the root type is; and the foreign array is embedded within the mashalling struct.
So we shouln't reach here.

Reply:
We don't reach here exactly, but there were several places where we had to cast to `void *` when marshalling foreign arrays that we weren't yet doing correctly.